lib/Dialog/Dialog.js

```diff
@@ -153,6 +165,12 @@ module.exports = kind(
 	*/
 	create: function () {
 		Popup.prototype.create.apply(this, arguments);
+		if(this.bodyComponents) {
+			this.$.message.destroy();
```

Comment:
The basic idea is not to create unnecessary components. But creating the components part is not handled by the dialog component, it's being taken care by enyo Popup and we are just providing `tools` not `components`. My previous commit in the this PR has a different approach. 
@aarontam  Can you please check it once. !? I can proceed with extending initComponents also but it will change the design of Dialog and seems a bit overhead for the component.
Thankyou.